src/sas/qtgui/convertUI.py

```diff
@@ -48,13 +51,17 @@ def file_in_newer(file_in, file_out):
     # simple comparison of modification time
     return in_stat.st_mtime >= out_stat.st_mtime
 
+
+args = sys.argv
+force_recreate = '-f' in args
+
 # look for .ui files
 for root, dirs, files in os.walk("."):
     for file in files:
         if file.endswith(".ui"):
             file_in = os.path.join(root, file)
             file_out = os.path.splitext(file_in)[0]+'.py'
-            if file_in_newer(file_in, file_out):
+            if force_recreate or file_in_newer(file_in, file_out):
```

Comment:
this conditional is evaluated three times. It may be easier to pre-evaluate it.
```
redo = force_recreate or file_in_newer(file_in, file_out)
...
if redo:
...
```

Reply:
`file_in_newer` passes different files for `file_in` and `file_out` every time it is run, so this would create an invalid check. I could probably incorporate the later checks into this loop. Let me look into that.